### ISSUE TITLE
Add profile fields and ability to update profile description/fields

### ISSUE THREAD

**Issue by TomCasavant** [ckolderup/postmarks#189]
I've got this setup in my fork, I can create a pull request if you have any guidance on how you'd prefer this to look

right now I've set it up so you can change it via the account.json:
```
{
  "username": "tom",
  "avatar": "https://cdn.glitch.global/8eaf209c-2fa9-4353-9b99-e8d8f3a5f8d4/postmarks-logo-white-small.png?v=1693610556689",
  "displayName": "Tom's Bookmarks",
  "description": "An ActivityPub bookmarking and sharing site built with <a href='https://github.com/ckolderup/postmarks' target='_blank' rel='nofollow noopener noreferrer me'>Postmarks</a>.",
  "attachment": [
    {
      "type": "PropertyValue",
      "name": "Maintainer",
      "value": "<a href='https://tomkahe.com/@tom' target='_blank' rel='nofollow noopener noreferrer me'>@tom@tomkahe.com</a>"
    },
    {
      "type": "PropertyValue",
      "name": "Source Code",
      "value": "<a href='https://github.com/tomcasavant/tom-postmarks' target='_blank' rel='nofollow noopener noreferrer me'>https://github.com/tomcasavant/tom-postmarks</a>"
    }
  ]
}
```

I'm not sure if you'd want this simplified a bit or not so you don't plug-in the type for each attachment (I'm not sure what other types of attachments are possible either). And then sending an update to every server is pretty straightforward you just send UPDATE with the PERSON entity from actorJson(). I don't have this automatic yet though, I would assume we'd want some sort of button on the admin page to trigger an update?

From postmarks site:
![image](https://github.com/ckolderup/postmarks/assets/7014115/c2311a9e-9bf7-4b77-87bf-62917eca2224)

Visual from mastodon:
![image](https://github.com/ckolderup/postmarks/assets/7014115/7efa50cb-c784-4240-9574-9a0637a5c7bd)

**Comment by andypiper** [comment]
Love it. Would be happy to test a PR.

**Comment by ckolderup** [comment]
Taking a look at the code now but as a comment on some of your questions/ideas first:

> I'm not sure if you'd want this simplified a bit or not so you don't plug-in the type for each attachment (I'm not sure what other types of attachments are possible either). 

I think this is fine for now; if I do find the space to return to things here I would like to work with @johnholdun on reviving the plan to move things out of config files and into the admin interface and so we'd be able to abstract away stuff like this in ways that don't expose them to the user while still planning for the future.

> And then sending an update to every server is pretty straightforward you just send UPDATE with the PERSON entity from actorJson(). I don't have this automatic yet though, I would assume we'd want some sort of button on the admin page to trigger an update?

What do you think about just doing it on server startup as a starting point? On Glitch that happens automatically when you edit the file anyway and in general the app is pretty fast to start up, so cycling the process/container/however people are running it outside of Glitch should be fairly unobtrusive, albeit mildly annoying. If we moved the config into the admin interface later as discussed above then there'd presumably be a save button to commit the info to the database and that could also trigger the update push, but server startup seems like a sensible kludgy thing to do for now.

**Comment by ckolderup** [comment]
oh, I just saw your comment in the pull request about not wanting to spam the followers with the profile update. I guess that's valid-- I get the sense there's a lot of noise in AP traffic in general, but I suppose it's good of us not to contribute to it further.

**Comment by TomCasavant** [comment]
> What do you think about just doing it on server startup as a starting point? On Glitch that happens automatically when you edit the file anyway and in general the app is pretty fast to start up, so cycling the process/container/however people are running it outside of Glitch should be fairly unobtrusive, albeit mildly annoying.

> I get the sense there's a lot of noise in AP traffic in general

You're probably right that it doesn't matter that much. I think at the time I was in the mindset that *I* was editing a lot of the files in glitch which meant it was sending out a profile update w/ every updated line. But most people won't be doing that so those updates probably wouldn't go out unless they were actually updating the profile.

**Comment by ckolderup** [comment]
this is true-- and if you're working in a live postmarks instance on Glitch with public followers (brave!) and you temporarily want to disable the server restarting with every edit of that file, you can do so using the `watch.json` file ([here's an example in the hello-eleventy project](https://glitch.com/edit/#!/glitch-hello-eleventy?path=watch.json%3A1%3A0))

**Comment by TomCasavant** [comment]
> you temporarily want to disable the server restarting with every edit of that file

Oh that's useful information, I think I was just editing a vanilla postmarks instance and testing w/ activitypub.academy so I wasn't worried about the updates being sent out at the time but I figured it'd be an issue when it got merged into actual postmarks instances
